foreshadow/utils/common.py

```diff
@@ -106,3 +107,27 @@ def configure_column_sharer(self, column_sharer):
         """
         if hasattr(self, "column_sharer"):
             self.column_sharer = column_sharer
+
+
+class CustomizeParamsMixin:
+    """Mixin that customizes params returned from get_params."""
+
+    def customize_serialization_params(self, params):
+        """Select params that are only necessary.
+
+        Args:
+            params: the params returned from get_params
+
+        Returns:
+            selected_params: params that are selected.
+
+        """
+        import inspect
+
+        init_params = inspect.signature(self.__init__).parameters
+        selected_params = {
```

Comment:
See above docstring comment, this could be:
`selected_params = {name: params.get('name', None) for name in init_params if name in params}`
Also, .pop removes the key from the dict and .get doesnt. I would prefer to use .get unless you explicitly want the former action to prevent future maintainers from running into weird bugs because they miss that .pop() was used instead of .get().